#!/bin/sh -e
## 01-cf.data.debian.dpatch by Luigi Gangitano <luigi@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Default configuration file for Debian.

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad squid-2.6.1~/src/cf.data.pre squid-2.6.1/src/cf.data.pre
--- squid-2.6.1~/src/cf.data.pre	2006-06-30 23:23:05.000000000 +0200
+++ squid-2.6.1/src/cf.data.pre	2006-07-03 14:48:08.000000000 +0200
@@ -320,12 +320,12 @@
 NAME: htcp_port
 IFDEF: USE_HTCP
 TYPE: ushort
-DEFAULT: 4827
+DEFAULT: 0
 LOC: Config.Port.htcp
 DOC_START
 	The port number where Squid sends and receives HTCP queries to
-	and from neighbor caches.  Default is 4827.  To disable use
-	"0".
+	and from neighbor caches.  To turn it on you want to set it 4827.
+	By default it is set to "0" (disabled).
 DOC_END
 
 
@@ -1611,6 +1611,10 @@
 	If append_domain is used, that domain will be added to
 	domain-local (i.e. not containing any dot character) host
 	definitions.
+
+NOCOMMENT_START
+hosts_file /etc/hosts
+NOCOMMENT_END
 DOC_END
 
 NAME: diskd_program
@@ -1823,7 +1827,7 @@
 
 	Then, set this line to something like
 
-	auth_param basic program @DEFAULT_PREFIX@/libexec/ncsa_auth @DEFAULT_PREFIX@/etc/passwd
+  	auth_param basic program @DEFAULT_PREFIX@/lib/squid/ncsa_auth @DEFAULT_PREFIX@/etc/passwd
 
 	"children" numberofchildren
 	The number of authenticator processes to spawn. If you start too few 
@@ -1883,7 +1887,7 @@
 
 	Then, set this line to something like
 
-	auth_param digest program @DEFAULT_PREFIX@/libexec/digest_auth_pw @DEFAULT_PREFIX@/etc/digpass
+ 	auth_param digest program @DEFAULT_PREFIX@/lib/squid/digest_auth_pw @DEFAULT_PREFIX@/etc/digpass
 
 
 	"children" numberofchildren
@@ -1945,7 +1949,10 @@
 	By default, the ntlm authentication scheme is not used unless a
 	program is specified.
 
-	auth_param ntlm program /path/to/samba/bin/ntlm_auth --helper-protocol=squid-2.5-ntlmssp
+	Note: If you're using Samba >= 3.0.2, please install the winbind
+	package and use the ntlm_auth helper from that package.
+
+	auth_param ntlm program /usr/bin/ntlm_auth --helper-protocol=squid-2.5-ntlmssp
 
 	"children" numberofchildren
 	The number of authenticator processes to spawn. If you start too few 
@@ -2751,7 +2758,8 @@
 acl manager proto cache_object
 acl localhost src 127.0.0.1/255.255.255.255
 acl to_localhost dst 127.0.0.0/8
-acl SSL_ports port 443 563
+acl SSL_ports port 443 563	# https, snews
+acl SSL_ports port 873		# rsync
 acl Safe_ports port 80		# http
 acl Safe_ports port 21		# ftp
 acl Safe_ports port 443 563	# https, snews
@@ -2762,6 +2770,10 @@
 acl Safe_ports port 488		# gss-http
 acl Safe_ports port 591		# filemaker
 acl Safe_ports port 777		# multiling http
+acl Safe_ports port 631		# cups
+acl Safe_ports port 873		# rsync
+acl Safe_ports port 901		# SWAT
+acl purge method PURGE
 acl CONNECT method CONNECT
 NOCOMMENT_END
 DOC_END
@@ -2881,6 +2893,9 @@
 # Only allow cachemgr access from localhost
 http_access allow manager localhost
 http_access deny manager
+# Only allow purge requests from localhost
+http_access allow purge localhost
+http_access deny purge
 # Deny requests to unknown ports
 http_access deny !Safe_ports
 # Deny CONNECT to other than SSL ports
@@ -2898,6 +2913,7 @@
 # be allowed
 #acl our_networks src 192.168.1.0/24 192.168.2.0/24
 #http_access allow our_networks
+http_access allow localhost
 
 # And finally deny all other access to this proxy
 http_access deny all
@@ -3230,12 +3246,12 @@
 
 NAME: cache_effective_user
 TYPE: string
-DEFAULT: nobody
+DEFAULT: proxy
 LOC: Config.effectiveUser
 DOC_START
 	If you start Squid as root, it will change its effective/real
 	UID/GID to the user specified below.  The default is to change
-	to UID to nobody.  If you define cache_effective_user, but not
+	to UID to proxy.  If you define cache_effective_user, but not
 	cache_effective_group, Squid sets the GID to the effective
 	user's default group ID (taken from the password file) and
 	supplementary group list from the from groups membership of
@@ -3245,7 +3261,7 @@
 
 NAME: cache_effective_group
 TYPE: string
-DEFAULT: none
+DEFAULT: proxy
 LOC: Config.effectiveGroup
 DOC_START
 	If you want Squid to run with a specific GID regardless of
@@ -3421,7 +3437,7 @@
 
 NAME: logfile_rotate
 TYPE: int
-DEFAULT: 10
+DEFAULT: 0
 LOC: Config.Log.rotateNumber
 DOC_START
 	Specifies the number of logfile rotations to make when you
@@ -3437,6 +3453,9 @@
 	purposes, so -k rotate uses another signal.  It is best to get
 	in the habit of using 'squid -k rotate' instead of 'kill -USR1
 	<pid>'.
+
+	Note2, for Debian/Linux the default of logfile_rotate is
+	zero, since it includes external logfile-rotation methods.
 DOC_END
 
 
@@ -4082,12 +4101,15 @@
 NAME: snmp_port
 TYPE: ushort
 LOC: Config.Port.snmp
-DEFAULT: 3401
+DEFAULT: 0
 IFDEF: SQUID_SNMP
 DOC_START
 	Squid can now serve statistics and status information via SNMP.
 	By default it listens to port 3401 on the machine. If you don't
 	wish to use SNMP, set this to "0".
+
+	Note: on Debian/Linux, the default is zero - you need to
+	set it to 3401 to enable it.
 DOC_END
 
 NAME: snmp_access
